c2rust-transpile/src/translator/builtins.rs

```diff
@@ -375,6 +375,16 @@ impl<'c> Translation<'c> {
             "__builtin_ia32_shufpd" => self.convert_simd_builtin(ctx, "_mm_shuffle_pd", args),
             "__builtin_ia32_shufps256" => self.convert_simd_builtin(ctx, "_mm256_shuffle_ps", args),
             "__builtin_ia32_shufpd256" => self.convert_simd_builtin(ctx, "_mm256_shuffle_pd", args),
+            "__builtin_ia32_pause" => {
+                let fn_name = "_mm_pause";
+                self.import_simd_function(fn_name)?;
+                let ident = mk().ident_expr(fn_name);
+                let call = mk().call_expr(ident, vec![]);
+                Ok(WithStmts::new(
+                    vec![mk().semi_stmt(call)],
+                    self.panic_or_err("No value for _mm_pause"),
+                ))
+            }
```

Comment:
This isn't really a SIMD function, so I'd move it out of this group of builtins.

Reply:
That said, I'd prefer to emit the safe and cross-platform `core::hint::spin_loop` instead.  It directly calls `_mm_pause` on x86/x86_64, but also supports other platforms as well: https://github.com/rust-lang/rust/blob/d6120810e56387730b2e86115471354c8084ca4a/library/core/src/hint.rs#L269-L304.

Furthermore, while the x86/x86_64 `_mm_pause` is stable now, the arm/aarch64 equivalent `__yield` is not and requires `#![feature(stdarch_arm_hints)]` (`aarch64`) and `#![feature(stdarch_arm_neon_intrinsics)]` (`arm`).  `__builtin_ia32_pause` is obviously x86/x86_64 only, but I think it'd be better to support the more cross-platform version that makes implementing other architectures' equivalents a lot simpler.

Reply:
Discussed offline with @kkysen: I'm not convinced we should try to emit `spin_loop` here in the interest of portability for two reasons:

1. The `curl` code which needs this feature emits `yield` on the `aarch64` code path but it seems `spin_loop` lowers to `isb` on `aarch64` so now we've introduced a subtle codegen difference between the C and Rust version. This makes cross checking more difficult.

2. When the programmer emits x86-specific intrinsics, it is likely on code paths that are inherently target/toolchain/OS specific and, IMHO, it is not worth it to emit more portable intrinsic rather than doing something "less surprising".